src/utils/StepIcon.tsx

```diff
@@ -0,0 +1,77 @@
+import styled from "@emotion/styled";
+import React from "react";
+import { IcStep1, IcStep2, IcStep3, IcStep4, IcCheckTransparent, IcGreyCircle } from "../assets";
+
+interface StepIconProps {
+  step: 1 | 2 | 3 | 4;
+}
+
+const StepIcon: React.FC<StepIconProps> = ({ step }) => {
+  return (
+    <StepIconContainer>
+      {step === 1 && (
+        <>
+          <StepIconWrapper>
+            <IcStep1 />
+            <StepText>약관동의</StepText>
+          </StepIconWrapper>
+          <IcGreyCircle />
+          <IcGreyCircle />
+          <IcGreyCircle />
+        </>
+      )}
+      {step === 2 && (
+        <>
+          <IcCheckTransparent />
+          <StepIconWrapper>
+            <IcStep2 />
+            <StepText>메일인증</StepText>
+          </StepIconWrapper>
+          <IcGreyCircle />
+          <IcGreyCircle />
+        </>
+      )}
+      {step === 3 && (
+        <>
+          <IcCheckTransparent />
+          <IcCheckTransparent />
+          <StepIconWrapper>
+            <IcStep3 />
+            <StepText>정보입력</StepText>
+          </StepIconWrapper>
+          <IcGreyCircle />
+        </>
+      )}
+      {step === 4 && (
+        <>
+          <IcCheckTransparent />
+          <IcCheckTransparent />
+          <IcCheckTransparent />
+          <StepIconWrapper>
+            <IcStep4 />
+            <StepText>상세정보</StepText>
+          </StepIconWrapper>
+        </>
+      )}
+    </StepIconContainer>
+  );
+};
+
+const StepIconContainer = styled.div`
+  display: flex;
+  gap: 0.5rem;
+`;
+
+const StepIconWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const StepText = styled.span`
+  margin-top: 0.3rem;
+  ${({ theme }) => theme.fonts.description3};
+  color: ${({ theme }) => theme.colors.gray_3};
+`;
+
+export default StepIcon;
```

Comment:
step은 prop으로 받고 있고 내부에서 상태를 관리하고 있지도 않기 때문에 prop에만 의존적인 리렌더링을 하고 있어요.
StepIcon 컴포넌트는 step이 바뀌지 않는 한 변경되는 부분이 없어요. 그래서 step이 변경될때만 리렌더링이 발생하면 될 것 같네요!!

이렇게 메모이제이션을 해서 불필요한 리렌더링을 줄이면 좋을 것 같아요!!

```suggestion
export default memo(StepIcon);
```